pkg/devicetwin/dtcontext/dtcontext_test.go

```diff
@@ -0,0 +1,493 @@
+		wantErr    bool
+	}{
+		{
+			//Success Case
+			name: "PingTest",
+			dtc: &DTContext{
+				ModulesHealth: &sync.Map{},
+			},
+			moduleName: "ModuleA",
+			content:    "ping",
+			wantErr:    false,
+		},
+		{
+			//Failure Case
+			name: "StopTest",
+			dtc: &DTContext{
+				ModulesHealth: &sync.Map{},
+			},
+			moduleName: "ModuleB",
+			content:    "stop",
+			wantErr:    true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			dtc := &DTContext{
+				ModulesHealth: test.dtc.ModulesHealth,
+			}
+			if err := dtc.HeartBeat(test.moduleName, test.content); (err != nil) != test.wantErr {
+				t.Errorf("DTContext.HeartBeat() error = %v, wantErr %v", err, test.wantErr)
+			}
+		})
+	}
+}
+
+//TestGetMutex is function to test GetMutex().
+func TestGetMutex(t *testing.T) {
+	dtc := &DTContext{
+		DeviceMutex: &sync.Map{},
+	}
+	var testMutex *sync.Mutex
+	dtc.DeviceMutex.Store("DeviceB", "")
+	dtc.DeviceMutex.Store("DeviceA", testMutex)
+	tests := []struct {
+		name     string
+		want     *sync.Mutex
+		wantBool bool
+		deviceID string
+	}{
+		{
+			//Failure Case-No device present
+			name:     "UnknownDevice",
+			wantBool: false,
+			deviceID: "DeviceC",
+		},
+		{
+			//Failure Case-Device present but unable to get mutex
+			name:     "UnableToGetMutex",
+			wantBool: false,
+			deviceID: "DeviceB",
+		},
+		{
+			//Success Case
+			name:     "KnownDevice",
+			wantBool: true,
+			deviceID: "DeviceA",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got, gotBool := dtc.GetMutex(test.deviceID)
+			if !reflect.DeepEqual(got, test.want) {
+				t.Errorf("DTContext.GetMutex() got = %v, want = %v", got, test.want)
+				return
+			}
+			if gotBool != test.wantBool {
+				t.Errorf("DTContext.GetMutex() gotBool = %v, want = %v", gotBool, test.wantBool)
+				return
+			}
+
+		})
+	}
+}
+
+//TestLock is function to test Lock().
+func TestLock(t *testing.T) {
+	dtc := &DTContext{
+		Mutex:       &sync.Mutex{},
+		DeviceMutex: &sync.Map{},
+	}
+	var testMutex sync.Mutex
+	dtc.DeviceMutex.Store("DeviceA", &testMutex)
+	tests := []struct {
+		name     string
+		deviceID string
+		want     bool
+	}{
+		{
+			//Failure Case
+			name:     "UnknownDevice",
+			deviceID: "DeviceB",
+			want:     false,
+		},
+		{
+			//Success Case
+			name:     "KnownDevice",
+			deviceID: "DeviceA",
+			want:     true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if got := dtc.Lock(test.deviceID); got != test.want {
+				t.Errorf("DTContext.Lock() = %v, want %v", got, test.want)
+			}
+		})
+	}
+}
+
+func TestUnlock(t *testing.T) {
+	dtc := &DTContext{
+		Mutex:       &sync.Mutex{},
+		DeviceMutex: &sync.Map{},
+	}
+	// Creating a mutex variable and getting a lock over it.
+	var testMutex sync.Mutex
+	dtc.DeviceMutex.Store("DeviceA", &testMutex)
+	testMutex.Lock()
+	tests := []struct {
+		name     string
+		deviceID string
+		want     bool
```

Comment:
rename to wantBool and follow similar convention in all test functions above and below